### DISCUSSION TITLE
Can't figure out how to use the action

### DISCUSSION THREAD

**Discussion by LFGaming** [lowlighter/metrics#161] [Q&A] [answered]
Hi,
I am new to the actions thing of github, so this could very well be something dumb I am doing wrong. But i copyed to code from the market place to use this metrics action.
This is my code I used:
```# This is a basic workflow to help you get started with Actions

name: CI

# Controls when the action will run. 
on:
  # Triggers the workflow on push or pull request events but only for the main branch
  push:
    branches: [ main ]
  pull_request:
    branches: [ main ]

  # Allows you to run this workflow manually from the Actions tab
  workflow_dispatch:

# A workflow run is made up of one or more jobs that can run sequentially or in parallel
jobs:
  # This workflow contains a single job called "build"
  build:
    # The type of runner that the job will run on
    runs-on: ubuntu-latest

    # Steps represent a sequence of tasks that will be executed as part of the job
    steps:
      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
      - uses: actions/checkout@v2

      # Runs a single command using the runners shell
      - name: Run a one-line script
        run: echo Hello, world!

      # Runs a set of commands using the runners shell
      - name: Run a multi-line script
        run: |
           echo Add other actions to build,
           echo test, and deploy your project.

      # Run metrics
      - name: GitHub metrics as SVG image
        uses: lowlighter/metrics@v3.3
```
I just used the standard action yml because then I knew for sure that the rest would work.
now i get these errors
![brave_2021-03-01_09-44-42](https://user-images.githubusercontent.com/38732192/109472746-ce3e9380-7a72-11eb-85f7-2c873b9b0ad2.png)
Could anyone help me with this? I have no idea what i should do... Thanks

Edit: 
I also got these errors.
![image](https://user-images.githubusercontent.com/38732192/109473272-62a8f600-7a73-11eb-9dd5-32f6e8d50876.png)

**Comment by LFGaming** [comment, marked as answer]
Edit:
I now used the website to make the action code.
I used this: 
```# Visit https://github.com/lowlighter/metrics/blob/master/action.yml for full reference
name: Metrics
on:
  # Schedule updates (each hour)
  schedule: [{cron: "0 * * * *"}]
  # Lines below let you run workflow manually and on each commit
  push: {branches: ["master", "main"]}
  workflow_dispatch:
jobs:
  github-metrics:
    runs-on: ubuntu-latest
    steps:
      - uses: lowlighter/metrics@latest
        with:
          # Your GitHub token
          token: ${{ secrets.METRICS_TOKEN }}
          # GITHUB_TOKEN is a special auto-generated token restricted to current repository, which is used to push files in it
          committer_token: ${{ secrets.GITHUB_TOKEN }}

          # Options
          user: LFGaming
          template: classic
          base: header, activity, community, repositories, metadata
          config_timezone: Europe/Amsterdam
          plugin_introduction: yes
          plugin_introduction_title: yes
          plugin_isocalendar: yes
          plugin_isocalendar_duration: half-year
          plugin_lines: yes
```
But I get these errors:
![brave_2021-03-01_10-12-58](https://user-images.githubusercontent.com/38732192/109476129-d6003700-7a76-11eb-945e-42ec1d7aa286.png)
![brave_2021-03-01_10-13-37](https://user-images.githubusercontent.com/38732192/109476139-d8fb2780-7a76-11eb-8fbd-62a6fc9be42a.png)
What am i doing wrong this time?

**Reply by LFGaming** [reply]
I somehow fixed it... Aparently my personal token thingy wasn't correct.

**Reply by lowlighter** [reply]
Glad you figured out 🙂 !

**Reply by LFGaming** [reply]
I don't know if i've done anything wrong, but there are a lot of actions that are queued but after that there are some that just work, and the queued once i can't cancel. is this a problem? and did i do something stupid?
![brave_2021-03-02_10-20-05](https://user-images.githubusercontent.com/38732192/109626527-f0064c00-7b40-11eb-9c88-338486846509.png)

**Reply by lowlighter** [reply]
Hum I'm not sure, but this may be related to [this thread](https://github.community/t/cant-cancel-github-action-workflow/16060/42).

Seems that GitHub have some issues atm with workflow runs in invalid states which were actually never queued succesfully.
From their answers, it seems that they setup a cron job to clean this, but it's running each 72h so I guess that you just need to wait 😕

**Reply by LFGaming** [reply]
ah okay, well at least it still works :). shall I then close this discussion? and if so how XD

**Reply by lowlighter** [reply]
Yeah ah ah x)
I selected your answer to close it 🙂 

Thanks for using metrics!
